### PR TITLE
🐛 Attempt to fix duplicate costs

### PR DESCRIPTION
**Pull request by davidreneuw**
# Pull Request

## Description

Properly define equality and hash values for DailyServiceCost so that a hash set can properly disregard duplicates when querying Azure. I determined the problem to be caused by ProjectUsageScheduler, and not ProjectUsageUpdater by checking the costs blob we store in Azure. Therefore, the problem is with how we query Azure and save the costs. That said, I am not 100% certain this is the fix but I am pretty confident.

This PR is now dependent on #1390 

## Related Issues

https://dev.azure.com/DataSolutionsDonnees/FSDH%20SSC/_sprints/taskboard/FSDH%20Dev%20Team/FSDH%20SSC/3.21?workitem=7037

## Changes

- Added `Equals` and `GetHashCode` overrides on the DailyServiceCost object
- Added unit tests to catch this in the future 

## Testing

- All existing tests pass
- New unit tests to ensure this does not happen again 

## Checklist

- [x] Code follows dotnet coding standards
- [x] Documentation updated (if necessary)
- [x] Tests added/updated (if necessary)

### CODE REVIEW

**Review by yjmrobert — DISMISSED** [review]

**Comment by yjmrobert** [comment]
Just waiting on a unit test to cover this bug

**Comment by yjmrobert** [comment]
@davidreneuw what's the status on this?

**Comment by davidreneuw** [comment]
> @davidreneuw what's the status on this?

Unit tests added and PR ready for re-review. Note this depends on #1390

File: Portal/src/Datahub.Application/Services/Cost/IWorkspaceCostManagementService.cs [hunk 1]
```diff
@@ -102,10 +102,15 @@ public class DailyServiceCost
         public string ResourceGroupName { get; set; } = String.Empty;
```
Comment (ErikApption) [inline Portal/src/Datahub.Application/Services/Cost/IWorkspaceCostManagementService.cs]: DailyServiceCost should prob move outside IWorkspaceCostManagementService - I don't mind having multiple classes in a single file, but maybe we can move these small helpers into SupportingObjects.cs or so

**Review by ErikApption — DISMISSED** [review]

File: Portal/src/Datahub.Application/Services/Cost/IWorkspaceCostManagementService.cs (hunk 1 above)
Comment (davidreneuw) [reply Portal/src/Datahub.Application/Services/Cost/IWorkspaceCostManagementService.cs]: I moved DailyServiceCost and its extensions into a different file!

**Comment by davidreneuw** [comment]
This is ready for merge once the pipelines have run

**Review by ErikApption — APPROVED** [review]

**Pull request merged**